bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatAccountHandler.java

```diff
@@ -0,0 +1,449 @@
+                if (rsp == null) {
+                    return (T) null;
+                } else if (rsp.errorCode == 0) {
+                    return (T) rsp;
+                } else {
+                    throw new MillheatCommunicationException(req, rsp);
+                }
+            } else {
+                throw new MillheatCommunicationException(
+                        "Error sending request to Millheat server. Server responded with " + contentResponse.getStatus()
+                                + " and payload " + responseJson);
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new MillheatCommunicationException("Error sending request to Millheat server", e);
+        }
+    }
+
+    protected MillheatModel refreshModel() throws MillheatCommunicationException {
+        MillheatModel model = new MillheatModel(System.currentTimeMillis());
+        GetHomesResponse homesRsp = sendLoggedInRequest(new GetHomesRequest(), GetHomesResponse.class);
+        for (HomeDTO dto : homesRsp.homes) {
+            model.addHome(new Home(dto));
+        }
+        for (Home home : model.getHomes()) {
+            SelectRoomByHomeResponse roomRsp = sendLoggedInRequest(
+                    new SelectRoomByHomeRequest(Long.parseLong(home.getId()), home.getTimezone()),
+                    SelectRoomByHomeResponse.class);
+            for (RoomDTO dto : roomRsp.rooms) {
+                home.addRoom(new Room(dto, home));
+            }
+
+            for (Room room : home.getRooms()) {
+                SelectDeviceByRoomResponse deviceRsp = sendLoggedInRequest(
+                        new SelectDeviceByRoomRequest(Long.parseLong(room.getId()), home.getTimezone()),
+                        SelectDeviceByRoomResponse.class);
+                for (DeviceDTO dto : deviceRsp.devices) {
+                    room.addHeater(new Heater(dto, room));
+                }
+            }
+            GetIndependentDevicesByHomeResponse independentRsp = sendLoggedInRequest(
+                    new GetIndependentDevicesByHomeRequest(Long.parseLong(home.getId()), home.getTimezone()),
+                    GetIndependentDevicesByHomeResponse.class);
+            for (DeviceDTO dto : independentRsp.devices) {
+                home.addHeater(new Heater(dto));
+            }
+        }
+        return model;
+    }
+
+    /**
+     * Stops this thing's polling future
+     */
+    @SuppressWarnings("null")
+    private void stopPolling() {
+        if (statusFuture != null && !statusFuture.isCancelled()) {
+            statusFuture.cancel(true);
+            statusFuture = null;
+        }
+    }
+
+    public void updateModelFromServerAndUpdateThingStatus() {
+        if (allowModelUpdate()) {
+            boolean success = false;
+            int retriesLeft = 2;
+            while (retriesLeft > 0) {
+                retriesLeft--;
+                try {
+                    model = refreshModel();
```

Comment:
Do you do this without any delays? I think that rescheduling might make more sense